devito/function.py

```diff
@@ -169,21 +170,21 @@ def _allocate_memory(func):
         def wrapper(self):
             if self._data is None:
                 debug("Allocating memory for %s%s" % (self.name, self.shape_allocated))
-                self._data = Data(self.shape_allocated, self.indices, self.dtype,
-                                  self._decomposition, self._allocator)
+                self._data = Data(self.shape_allocated, self.dtype,
+                                  modulo=self._mask_modulo, allocator=self._allocator)
                 if self._first_touch:
-                    first_touch(self)
+                    assign(self, 0)
```

Comment:
Does this actually fix the first_touch issues we had? Don't rememberwhat the issue was